File: longest_common_prefix/LongestCommonPrefix.java
class LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs) {
        
        StringBuilder sb = new StringBuilder("");
        try{
            for(int i = 0; i < strs[0].length(); i++){
                for(int j = 0; j < strs.length; j++){
                    if(strs[0].charAt(i)!=strs[j].charAt(i)) return sb.toString();
                }
                sb.append(strs[0].charAt(i));
            }
            return sb.toString();
        }
        catch(Exception e){
            return sb.toString();
        }
        
    }
}

File: longest_common_prefix/longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        str_sort = lambda x: len(x)
        strs.sort(key=str_sort)
        s = ""
        for n in range(len(strs[0])):
            for i in range(len(strs)):
                if strs[0][n]!=strs[i][n]:
                    return s
            s+=strs[0][n]
        return s
        